src/orion/core/worker/trial.py

```diff
@@ -294,44 +294,35 @@ def gradient(self):
         """
         return self._fetch_one_result_of_type('gradient')
 
-    def _repr_values(self, values, sep=','):
-        """Represent with a string the given values."""
-        return sep.join(map(lambda value: "{0.name}:{0.value}".format(value), values))
-
-    def params_repr(self, sep=',', ignore_fidelity=False):
-        """Represent with a string the parameters contained in this `Trial` object."""
-        if ignore_fidelity:
-            params = [x for x in self._params if x.type != 'fidelity']
-        else:
-            params = self._params
-        return self._repr_values(params, sep)
-
     @property
     def hash_name(self):
         """Generate a unique name with an md5sum hash for this `Trial`.
 
         .. note:: Two trials that have the same `params` must have the same `hash_name`.
         """
-        if not self._params and not self.experiment:
-            raise ValueError("Cannot distinguish this trial, as 'params' or 'experiment' "
-                             "have not been set.")
-        params_repr = self.params_repr()
-        experiment_repr = str(self.experiment)
-        lie_repr = self._repr_values([self.lie]) if self.lie else ""
-        return hashlib.md5((params_repr + experiment_repr + lie_repr).encode('utf-8')).hexdigest()
+        return self._hash(ignore_fidelity=False)
 
     @property
     def hash_params(self):
         """Generate a unique param md5sum hash for this `Trial`.
 
         .. note:: The params contributing to the hash do not include the fidelity.
         """
+        return self._hash(ignore_fidelity=True)
+
+    def _hash(self, ignore_fidelity=False):
```

Comment:
Maybe this should be a public static method well since we plan on using it for the Track backend.

Reply:
ok